zerver/views/realm.py

```diff
@@ -62,6 +64,9 @@ def update_realm(request, user_profile, name=REQ(validator=check_string, default
     if create_stream_by_admins_only is not None and realm.create_stream_by_admins_only != create_stream_by_admins_only:
         do_set_realm_create_stream_by_admins_only(realm, create_stream_by_admins_only)
         data['create_stream_by_admins_only'] = create_stream_by_admins_only
+    if add_emoji_by_admins_only is not None and realm.add_emoji_by_admins_only != add_emoji_by_admins_only:
+        do_set_realm_add_emoji_by_admins_only(realm, add_emoji_by_admins_only)
```

Comment:
We can probably get some test coverage on this logic.  Can you look for tests that deal with some of the other settings in this method to find models?  (Maybe this is already covered, but it's not obvious from the diff.  You can run ./tools/test-backend --coverage to get coverage reports.)